numpy/core/records.py

```diff
@@ -773,7 +773,40 @@ def fromrecords(recList, dtype=None, shape=None, formats=None, names=None,
 def fromstring(datastring, dtype=None, shape=None, offset=0, formats=None,
                names=None, titles=None, aligned=False, byteorder=None):
     """Create a (read-only) record array from binary data contained in
-    a string"""
+    a string
+
+    Parameters
+    ----------
+    datastring : str
```

Comment:
```suggestion
    datastring : bytes-like
```
Based on the fact that `datastring` is used with the `buf` kwarg of `recarray`, I think this actually has to be some form of buffer. Of course, this makes  the `fromstring` name confusing. Maybe someone with more experience in how this was handled for e.g. `np.frombuffer` can offer better insight.